src/lints/pub_function_changed_abi.ron

```diff
@@ -0,0 +1,58 @@
+SemverQuery(
+    id: "pub_function_changed_abi",
+    human_readable_name: "pub function changed ABI",
+    description: "A public function changed its external ABI.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/external-blocks.html#abi"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+                        
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                        
+                        abi_: abi {
+                            name @output @tag
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                        
+                        abi @fold @transform(op: "count")  @filter(op: "=", value: ["$zero"]) {
+                            name @filter(op: "=", value: ["%name"])
+                        }
+
+                        new_abi_: abi {
+                            name @output
+                        }
```

Comment:
Functions are guaranteed to have exactly one ABI: we see this in the Trustfall schema where the ABI edge is defined `abi: FunctionAbi!` meaning exactly one neighboring vertex.

That makes this combination of query clauses equivalent to:
```suggestion
                        new_abi_: abi {
                            name @filter(op: "!=", value: ["%name"]) @output
                        }
```

In the original, we say a combination of "there is no matching ABI" + "get the actual ABI here." In the revised version I'm suggesting, we just say "the ABI that exists is not a match." This is the same thing when there's exactly one ABI, no more and no less. If there could be zero or more than one ABI, that wouldn't be equivalent and we'd need a structure like the original one here.